google-cloud-firestore/lib/google/cloud/firestore/client.rb

```diff
@@ -166,15 +167,8 @@ def col_group collection_id
             raise ArgumentError, "Invalid collection_id: '#{collection_id}', " \
               "must not contain '/'."
           end
-          query = Google::Cloud::Firestore::V1::StructuredQuery.new(
-            from: [
-              Google::Cloud::Firestore::V1::StructuredQuery::CollectionSelector.new(
-                collection_id: collection_id, all_descendants: true
-              )
-            ]
-          )
-
-          Query.start query, service.documents_path, self
+
+          CollectionGroup.from_collection_id service.documents_path, collection_id, self
```

Comment:
@schmidt-sebastian Like googleapis/java-firestore#202, this PR introduces a new subclass of `Query`: `CollectionGroup`. However, given the current implementation of `Query`, adding a clause such as `order by` to a collection group instance returns a new object that is of type `Query` and therefore cannot provide partitions. Is this incorrect?

```rb
    collection_group = firestore.col_group(collection_id)
    collection_group.class #=> Google::Cloud::Firestore::CollectionGroup
    collection_group = collection_group.order_by("foo")
    collection_group.class #=> Google::Cloud::Firestore::Query
    collection_group.partitions(3) #=> NoMethodError: undefined method `partitions' for #<Google::Cloud::Firestore::Query
```

Reply:
That is correct. Partition queries only support CollectionGroup queries with no constraints.

Reply:
@schmidt-sebastian Thank you for the confirmation. In that case, I believe this PR is ready for review.